### PR TITLE
Mpesa, ghana, uganda, card validation moved to presenter

### PR DESCRIPTION
**Pull request by mikelis135**
Mpesa, ghana, uganda, card validation moved to presenter

### CODE REVIEW

**Comment by mikelis135** [comment]
Mpesa, ghana, uganda, account, card validation moved to presenter

File: raveandroid/src/main/java/com/flutterwave/raveandroid/card/CardFragment.java [hunk 1]
```diff
@@ -102,80 +97,141 @@ public CardFragment() {
...
-            @Override
-            public void onClick(View v) {
-                presenter.onSavedCardsClicked(ravePayInitializer.getEmail());
-            }
-        });
-
         cardExpiryTv.addTextChangedListener(new ExpiryWatcher());
 
-        payButton.setOnClickListener(this);
+        setOnClickListeners();
```
Comment (biodunalfet) [inline raveandroid/src/main/java/com/flutterwave/raveandroid/card/CardFragment.java]: rename to setListeners()

File: raveandroid/src/main/java/com/flutterwave/raveandroid/card/CardFragment.java [hunk 2]
```diff
@@ -102,80 +97,141 @@ public CardFragment() {
...
         presenter.checkForSavedCards(ravePayInitializer.getEmail());
 
-        savedCardBtn.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                presenter.onSavedCardsClicked(ravePayInitializer.getEmail());
-            }
-        });
-
         cardExpiryTv.addTextChangedListener(new ExpiryWatcher());
```
Comment (biodunalfet) [inline raveandroid/src/main/java/com/flutterwave/raveandroid/card/CardFragment.java]: Move this to setListeners()

File: raveandroid/src/main/java/com/flutterwave/raveandroid/card/CardFragment.java [hunk 3]
```diff
@@ -102,80 +97,141 @@ public CardFragment() {
...
-                return "";
-            }
-        };
+        initializeViews();
 
-        Pattern pattern = Pattern.compile("()PCI-DSS COMPLIANT");
-        Linkify.addLinks(pcidss_tv, pattern, "https://www.pcisecuritystandards.org/pci_security/", null, filter);
+        pcidss_tv.setMovementMethod(LinkMovementMethod.getInstance());
 
         presenter.checkForSavedCards(ravePayInitializer.getEmail());
```
Comment (biodunalfet) [inline raveandroid/src/main/java/com/flutterwave/raveandroid/card/CardFragment.java]: remove this

**Pull request merged**